Library.Infrastructure/Repositories/BookRepository.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Library.Domain.Interfaces;
+using Library.Domain.Models;
+using Library.Domain.Data;
+
+namespace Library.Infrastructure.Repositories;
+
+/// <summary>
+/// Репозиторий с CRUD-операциями для Book.
+/// Использует данные из DataSeed.
+/// </summary>
+public class BookRepository: IRepositories<Book, int>
+{
+    private readonly List<Book> _books;
+    private int _maxId;
+
+    /// <summary>
+    /// Конструктор для репозитория, использует начальные данные из DataSeed 
+    /// и определяе максимальный текущий ID, для добавления новых объктов.
+    /// </summary>
+    public BookRepository()
+    {
+        _books = DataSeed.Books;
```

Comment:
Таким образом вы не создаете новый список а сохраняете ссылку на существующий. Это может быть не безопасно. Лучше бы скопировать значения из списка создав новый список на основе значения того списка. Аналогично по коду в других местах.

```suggestion
        _books = [.. DataSeed.Books];
```